src/ScheduledTasks/Entity/Schedule.cs

```diff
@@ -0,0 +1,348 @@
+            }
+
+            Verify.NotNull(this.State.ScheduleConfiguration, nameof(this.State.ScheduleConfiguration));
+
+            // Transition to Paused state
+            this.State.Status = ScheduleStatus.Paused;
+            this.State.NextRunAt = null;
+            this.State.RefreshScheduleRunExecutionToken();
+
+            this.logger.PausedSchedule(this.State.ScheduleConfiguration.ScheduleId);
+        }
+        catch (Exception ex)
+        {
+            this.logger.ScheduleOperationError(this.State.ScheduleConfiguration?.ScheduleId ?? string.Empty, nameof(this.PauseSchedule), "Failed to pause schedule", ex);
+            throw;
+        }
+    }
+
+    /// <summary>
+    /// Resumes the schedule.
+    /// </summary>
+    /// <param name="context">The task entity context.</param>
+    /// <exception cref="InvalidOperationException">Thrown when the schedule is not paused.</exception>
+    public void ResumeSchedule(TaskEntityContext context)
+    {
+        try
+        {
+            if (!this.CanTransitionTo(nameof(this.ResumeSchedule), ScheduleStatus.Active))
+            {
+                throw new ScheduleInvalidTransitionException(this.State.ScheduleConfiguration?.ScheduleId ?? string.Empty, this.State.Status, ScheduleStatus.Active, nameof(this.ResumeSchedule));
+            }
+
+            Verify.NotNull(this.State.ScheduleConfiguration, nameof(this.State.ScheduleConfiguration));
+
+            this.State.Status = ScheduleStatus.Active;
+            this.State.NextRunAt = null;
+            this.logger.ResumedSchedule(this.State.ScheduleConfiguration.ScheduleId);
+
+            // compute next run based on startat and interval
+            context.SignalEntity(new EntityInstanceId(nameof(Schedule), this.State.ScheduleConfiguration.ScheduleId), nameof(this.RunSchedule), this.State.ExecutionToken);
+        }
+        catch (Exception ex)
+        {
+            this.logger.ScheduleOperationError(this.State.ScheduleConfiguration?.ScheduleId ?? string.Empty, nameof(this.ResumeSchedule), "Failed to resume schedule", ex);
+            throw;
+        }
+    }
+
+    /// <summary>
+    /// Runs the schedule based on the defined configuration.
+    /// </summary>
+    /// <param name="context">The task entity context.</param>
+    /// <param name="executionToken">The execution token for the schedule.</param>
+    /// <exception cref="InvalidOperationException">Thrown when the schedule is not active or interval is not specified.</exception>
+    public void RunSchedule(TaskEntityContext context, string executionToken)
+    {
+        if (this.State.Status == ScheduleStatus.Uninitialized)
+        {
+            // this signal is no longer useful since the schedule has been deleted.
+            this.State = null!; // delete again, otherwise an uninitialized schedule will stick around
```

Comment:
Looking at this use of `null!`, shouldn't we make `this.State` to be nullable? I'm worried about unintended `NullReferenceExceptions`. For example, if you call `RunSchedule` twice in a row, won't hat result in a NRE?